File: source java/organism/animals/CyberSheep.java
package organism.animals;

import Interfaces.Requirements;
import java.awt.Color;
import organism.Organism;
import organism.manager.Organisms;
import organism.plants.SosnowskysHogweed;
import utilities.position.Position;
import utilities.savingsystem.Loader;
import virtualworld.World;

public class CyberSheep extends Sheep
{
    private static final int STRENGTH = 11;
    private static final int INITIATIVE = 4;
    
    public CyberSheep(Position position, World world) {
        super(STRENGTH, INITIATIVE, position, world);
    }
    
    public CyberSheep(Loader loader, World world, Position position)
    {
        super(loader, world, position);
    }

    @Override
    public void Act()
    {
        Position sosnowskysPosition = GetWorld().FindNearestPosition(GetPosition(), new Requirements()
        {
            @Override
            public boolean IsOk(Organism organism)
            {
                return (organism instanceof SosnowskysHogweed);
            }
        });
        
        if(sosnowskysPosition == null)
            super.Act();
        else
            GoTo(GetPosition().GetNeighbouringCloserPosition(sosnowskysPosition));
    }
    
    @Override
    public Color GetColor()
    {
        return Color.blue.brighter();
    }
    
    @Override
    public Organisms GetEnum()
    {
        return Organisms.CYBERSHEEP;
    }
    
    @Override
    public String toString()
    {
        return "Cybersheep";
    }
    
    
}

File: source python/Organisms/Animals/CyberSheep.py
from Application.Organisms.Animals.Animal import Animal
from Application.Utilities import Colors
from Application.Organisms.Plants.SosnowskysHogweed import SosnowskysHogweed


class CyberSheep(Animal):
    STRENGTH = 11
    INITIATIVE = 4

    def __init__(self, position, world, **kwargs):
        super().__init__(CyberSheep.STRENGTH, CyberSheep.INITIATIVE, position, world, **kwargs)

    def Act(self):
        class CyberSheepRequirements:
            def IsOk(self, organism):
                return isinstance(organism, SosnowskysHogweed)

        sosnowskysPosition = self.GetWorld().FindNearestPosition(self.GetPosition(), CyberSheepRequirements())
        if sosnowskysPosition is None:
            return super().Act()
        else:
            self._GoTo(self.GetPosition().GetNeighbouringCloserPosition(sosnowskysPosition))
    
    def GetColor(self):
        return Colors.PINK

    @staticmethod
    def __str__():
        return "Cybersheep"

    def GetLetter(self):
        return "C"
